From e7b57fa1c49f821fee74d617b855fb35dba76069 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 16 Jul 2014 20:42:09 -0400
Subject: [md] dm: use RCU_INIT_POINTER instead of rcu_assign_pointer in __unbind

Message-id: <1405543364-20901-6-git-send-email-snitzer@redhat.com>
Patchwork-id: 85889
O-Subject: [RHEL7.1 PATCH 05/40] dm: use RCU_INIT_POINTER instead of rcu_assign_pointer in __unbind
Bugzilla: 1117872
RH-Acked-by: Mikulas Patocka <mpatocka@redhat.com>
RH-Acked-by: Heinz Mauelshagen <heinzm@redhat.com>

BZ: 1117872

Upstream commit 9cdb8520049629271ad411ac91ab1bea3e1cfa2b
Author: Monam Agarwal <monamagarwal123@gmail.com>
Date:   Sun Mar 23 23:58:27 2014 +0530

    dm: use RCU_INIT_POINTER instead of rcu_assign_pointer in __unbind

    Replace rcu_assign_pointer(p, NULL) with RCU_INIT_POINTER(p, NULL).

    The rcu_assign_pointer() ensures that the initialization of a structure
    is carried out before storing a pointer to that structure.  And in the
    case of the NULL pointer, there is no structure to initialize.  So,
    rcu_assign_pointer(p, NULL) can be safely converted to
    RCU_INIT_POINTER(p, NULL).

    Signed-off-by: Monam Agarwal <monamagarwal123@gmail.com>
    Signed-off-by: Mike Snitzer <snitzer@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm.c b/drivers/md/dm.c
index 3bc36c0..ab7f3b2 100644
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -2296,7 +2296,7 @@ static struct dm_table *__unbind(struct mapped_device *md)
   return NULL;
 
  dm_table_event_callback(map, NULL, NULL);
- rcu_assign_pointer(md->map, NULL);
+ RCU_INIT_POINTER(md->map, NULL);
  dm_sync_table(md);
 
  return map;
-- 
1.7.1